### DISCUSSION TITLE
Should we remove streamedAmountOf from Linear/Dynamic/Tranched interfaces?

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/evm-monorepo#856] [General]
After this PR https://github.com/sablier-labs/v2-core/pull/813, the only duplicated getter function that remains in the interfaces is `streamedAmountOf`. Its presence in the unique interface is due to the differing natspec comments between them:

- [ISablierV2LockupLinear](https://github.com/sablier-labs/v2-core/blob/1b2ee6df156b156799a4e3bd680b726079b018fb/src/interfaces/ISablierV2LockupLinear.sol#L62-L74) 
- [ISablierV2LockupDynamic](https://github.com/sablier-labs/v2-core/blob/1b2ee6df156b156799a4e3bd680b726079b018fb/src/interfaces/ISablierV2LockupDynamic.sol#L66-L79)
- [ISablierV2LockupTranched](https://github.com/sablier-labs/v2-core/blob/1b2ee6df156b156799a4e3bd680b726079b018fb/src/interfaces/ISablierV2LockupTranched.sol#L66-L77)

Now, we are simply overriding the function implemented in `SablierV2Lockup`:

https://github.com/sablier-labs/v2-core/blob/1b2ee6df156b156799a4e3bd680b726079b018fb/src/SablierV2LockupLinear.sol#L128

I was considering whether it would be better to remove the function from the Linear/Dynamic/Tranched interfaces and move the natspec to the bespoke contract at `_calculateStreamedAmount`?

<details>
<summary> Example Linear </summary>

```solidity
    /// @inheritdoc SablierV2Lockup
    ///
    /// @dev When the stream is warm, the streaming function is:
    ///
    /// $$
    /// f(x) = x * d + c
    /// $$
    ///
    /// Where:
    ///
    /// - $x$ is the elapsed time divided by the stream's total duration.
    /// - $d$ is the deposited amount.
    /// - $c$ is the cliff amount.
    function _calculateStreamedAmount(uint256 streamId) internal view override returns (uint128) {
         // -- implementation  --
    }
```

</details>

What do you guys say? @PaulRBerg  @smol-ninja

**Comment by andreivladbrg** [comment]
I've pushed a commit here https://github.com/sablier-labs/v2-core/commit/b4a264d199f6aaa62ea2c4fc1aecd91b2af3ca98 to help you understand better what I am referring to.

**Comment by smol-ninja** [comment]
I don't have anything against this idea. It looks better to me. I find the streaming functions comment more suitable for `_calculateStreamedAmount` than `streamedAmountOf` because the latter also depends on the stream status while the former is a direct implementation of the formula.

**Comment by andreivladbrg** [comment]
Implemented in https://github.com/sablier-labs/v2-core/pull/864
